src/compiler.ts

```diff
@@ -33,6 +34,16 @@ function compileSingle(elements: RegexElement): string {
     return elements;
   }
 
+  if (!('children' in elements)) {
```

Comment:
If you add `as const` to character compiles, would that enable you to drop the `('children' in elements)` check?

Reply:
Unfortunately no. TS does not like when string passed as a key is wider than what object keys are, even if you handle such case by checking for `undefined`.

Alternatively I can check if `elements.type` is one of the correct keys, but when I tried it before it was not narrowing `elements.type` after the if statement, so it required additional checks for `elementCompiler`, etc.

```
const isCharacterClassKey = (value: string): value is keyof typeof characterClasses => {
  return value in characterClasses;
};

if (isCharacterClassKey(elements.type)) {
    return characterClasses[elements.type];
}
```

<img width="961" alt="image" src="https://github.com/callstack/ts-regex/assets/22678765/47d576c8-5d2a-45a8-80fb-6040b15c60d0">

Reply:
check current version, I've solved it